MapboxDirections/OfflineDirections.swift

```diff
@@ -14,20 +14,22 @@ struct AvailableVersionsResponse: Codable {
 public protocol OfflineDirectionsProtocol {
     
     /**
-     Fetches the available versions.
+     Fetches the available offline routing tile versions and returns them in descending chronological order. The most recent version should typically be passed into `downloadTiles(in:version:completionHandler:)`.
+     
+     - parameter completionHandler: A closure of type `OfflineVersionsHandler` which will be called when the request completes
      */
-    @discardableResult
-    func fetchAvailableOfflineVersions(completionHandler: @escaping OfflineVersionsHandler) -> URLSessionDataTask
+    @objc(fetchAvailableOfflineVersionsWithCompletionHandler:)
+    func fetchAvailableOfflineVersions(completionHandler: @escaping OfflineVersionsHandler) -> ()
     
     /**
-     Initiates a download process of all tiles needed to provide routing within the given bounding box.
+     Downloads offline routing tiles of the given version within the given coordinate bounds using the shared URLSession. The tiles are written to disk at the location passed into the `completionHandler`.
      
-     - parameter coordinateBounds: The region of the pack to be downloaded.
-     - parameter version: The version of the pack to be downloaded.
-     - parameter completionHandler: Informs when the download is completed or failed. The offline pack may be moved from the temporary directory and to a persistent store at this point.
+     - parameter coordinateBounds: The bounding box
+     - parameter version: The version to download. Version is represented as a String (yyyy-MM-dd-x)
+     - parameter completionHandler: A closure of type `OfflineDownloaderCompletionHandler` which will be called when the request completes
      */
-    @discardableResult
-    func downloadTiles(in coordinateBounds: CoordinateBounds, version: OfflineVersion, session: URLSession?, completionHandler: @escaping OfflineDownloaderCompletionHandler) -> URLSessionDownloadTask
+    @objc(downloadTilesInCoordinateBounds:withVersion:completionHandler:)
+    func downloadTiles(in coordinateBounds: CoordinateBounds, version: OfflineVersion, completionHandler: @escaping OfflineDownloaderCompletionHandler) -> ()
```

Comment:
Either the argument label in Swift should be `with:` or the selector piece in Objective-C should be just `version:`.

Reply:
@1ec5 should we plan a major release for this?